rest/route/host_create.go

```diff
@@ -70,7 +70,40 @@ func (h *hostCreateHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	ids := []string{}
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	initialHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
```

Comment:
The query needs to check more host statuses than just "running". If the host is [any of these statuses](https://github.com/evergreen-ci/evergreen/blob/624ba08c8e07fc6c39f757cf5ce5d273a4288d3b/globals.go#L992-L997), then it counts as a host towards the `num_hosts` requested. This function only looks for hosts in the running state, but they're only "running" when they're actually up and running in EC2. We want to make sure that we have `num_hosts` hosts that are either in the process of coming up, or are already up.

Reply:
I left a recommendation for how to consolidate the two.